JetMETCorrections/Type1MET/plugins/MultShiftMETcorrDBInputProducer.cc

```diff
@@ -0,0 +1,175 @@
+  METCorParamsColl->validKeys( keys );
+
+
+  //get primary vertices
+  edm::Handle<edm::View<reco::Vertex> > hpv;
+  evt.getByToken( vertices_, hpv );
+  if(!hpv.isValid()) {
+    edm::LogError("MultShiftMETcorrDBInputProducer::produce") << "could not find vertex collection ";
+  }
+  std::vector<reco::Vertex> goodVertices;
+  for (unsigned i = 0; i < hpv->size(); i++) {
+    if ( (*hpv)[i].ndof() > 4 &&
+       ( fabs((*hpv)[i].z()) <= 24. ) &&
+       ( fabs((*hpv)[i].position().rho()) <= 2.0 ) )
+       goodVertices.push_back((*hpv)[i]);
+  }
+  int ngoodVertices = goodVertices.size();
+
+  edm::Handle<edm::View<reco::Candidate> > particleFlow;
+  evt.getByToken(pflow_, particleFlow);
+
+
+  //MM: loop over all constituent types and sum each correction
+  std::auto_ptr<CorrMETData> metCorr(new CorrMETData());
+  
+  double corx=0.;
+  double cory=0.;
+
+
+  // check DB
+  for ( std::vector<METCorrectorParametersCollection::key_type>::const_iterator ibegin = keys.begin(),
+	  iend = keys.end(), ikey = ibegin; ikey != iend; ++ikey ) {
+    if(mIsData)
+    {
+      if(!METCorParamsColl->isXYshiftData(*ikey) )
+	throw cms::Exception("MultShiftMETcorrDBInputProducer::produce")
+	  << "DB is not for Data. Set proper option: \"corrPfMetXYMultDB.isData\" !!\n";
+    }else{
+      if( METCorParamsColl->isXYshiftData(*ikey) )
+	throw cms::Exception("MultShiftMETcorrDBInputProducer::produce")
+	  << "DB is for Data. Set proper option: \"corrPfMetXYMultDB.isData\" !!\n";
+    }
+  }
+
+  for ( std::vector<METCorrectorParametersCollection::key_type>::const_iterator ibegin = keys.begin(),
+	  iend = keys.end(), ikey = ibegin; ikey != iend; ++ikey ) {
+
+    if( !mIsData){
+
+      if(mSampleType == "MC")
+        if(!METCorParamsColl->isXYshiftMC(*ikey)) continue;
+      if(mSampleType == "DY")
+        if(!METCorParamsColl->isXYshiftDY(*ikey)) continue;
+      if(mSampleType == "TTJets")
+        if(!METCorParamsColl->isXYshiftTTJets(*ikey)) continue;
+      if(mSampleType == "WJets")
+        if(!METCorParamsColl->isXYshiftWJets(*ikey)) continue;
+    }
+
+    std::string sectionName= METCorParamsColl->findLabel(*ikey);
+    METCorrectorParameters const & METCorParams = (*METCorParamsColl)[*ikey];
+
+    counts_ = 0;
+    sumPt_  = 0;
+
+    for (unsigned i = 0; i < particleFlow->size(); ++i) {
+      const reco::Candidate& c = particleFlow->at(i);
+      if (abs(c.pdgId())== translateTypeToAbsPdgId(reco::PFCandidate::ParticleType( METCorParams.definitions().ptclType() ))) {
+        if ((c.eta()>METCorParams.record(0).xMin(0)) and (c.eta()<METCorParams.record(0).xMax(0))) {
+          counts_ +=1;
+          sumPt_  +=c.pt();
+          continue;
+        }
+      }
+    }
+    double val(0.);
+
+    if ( METCorParams.definitions().parVar(0) ==0) {
+      val = counts_;
+    } 
+    if ( METCorParams.definitions().parVar(0) ==1) {
+      val = ngoodVertices; 
+    } 
+    if ( METCorParams.definitions().parVar(0) ==2) {
+      val = sumPt_;
+    }
```

Comment:
Set a variable to `METCorParams.definitions().parVar(0)` and then check in the `if` statements.
`int parVar = METCorParams.definitions().parVar(0);`

Add `else` to these `if` statements.